### PR TITLE
Fix/revoke mutual consent

### PR DESCRIPTION
**Pull request by Craigson**
- Added a `revokeConsent` function to `MutualConsent.sol`
- Added dedicated test file: `MutualConsent.t.sol`
- Addresses ticket: https://www.notion.so/Add-revokeConsent-function-to-mutualConsent-aea626bc794d43239c95eaaa4e30bda3

### CODE REVIEW

File: contracts/utils/MutualConsent.sol [hunk 1]
```diff
@@ -11,43 +11,94 @@ pragma solidity 0.8.9;
...
+            _reconstrucedMsgData.length < MIN_DATA_LENGTH_BYTES
+        ) {
+            revert UnsupportedMutualConsentFunction();
+        }
+
+        bytes32 hashToDelete = keccak256(
+            abi.encodePacked(_reconstrucedMsgData, msg.sender)
+        );
+
+        if (mutualConsents[hashToDelete] == address(0)) {
```
Comment (kibagateaux) [inline contracts/utils/MutualConsent.sol:70]: create var and save address to memory for gas savings

File: contracts/tests/MutualConsent.t.sol [hunk 2]
```diff
@@ -0,0 +1,457 @@
...
+        // succeed revoking consent
+        vm.expectEmit(true, true, false, true, address(line));
+        emit MutualConsentRevoked(borrower, expectedHash);
+        line.revokeConsent(msgData);
+        vm.stopPrank();
+
+        assertEq(line.mutualConsents(expectedHash), address(0));
+
+        // lender addCredit should create a new id
+        vm.startPrank(lender);
```
Comment (kibagateaux) [inline contracts/tests/MutualConsent.t.sol:175]: Can you create this as a second test `revoking_consent_must_delete_consent_hash`

Comment (kibagateaux) [reply contracts/tests/MutualConsent.t.sol:175]: Or i guess 2 new tests. 1 check that hash => address is null and 1 check that calling hash after revoking doesnt succeed (technically different stories)

Comment (Craigson) [reply contracts/tests/MutualConsent.t.sol:175]: @kibagateaux to test for the second condition, the call obviously succeeds, but it creates a new entry in the `MutualConsents` mapping and emits the `MutualConsentRegistered` event, which is the condition for passing the test.

**Pull request merged**